lib/request_overrider.js

```diff
@@ -307,41 +306,29 @@ function RequestOverrider(req, options, interceptors, remove) {
     if (interceptor.replyFunction) {
       const parsedRequestBody = parseJSONRequestBody(req, requestBody)
 
-      if (interceptor.replyFunction.length === 3) {
+      let fn = interceptor.replyFunction
+      if (fn.length === 3) {
```

Comment:
You could also consider checking if it is an AsyncFunction (and in that case not promisifying, regardless of the number of arguments):

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AsyncFunction

It ensures that an async reply function is awaited, even if the arguments have gotten messed up.

Reply:
I have reservations on this. An async function and a function returning a promise should be treated the same in my book, but there is not way to sniff for the later. 

With some quick testing, it looks like even if an async fn gets wrapped in `promisify` it will resolve the whole chain as desired.

Reply:
I agree with the principle that an async function and a promise should be treated the same.

A conflicting principle (in this case) is that the number of arguments to a promise-returning function should not affect its behavior. I see why we did that here, and when we lived in the world of callbacks, why it would be necessary. Though going forward ideally we wouldn't care about the number of arguments to the function. My goal is to avoid surprises when someone makes a mistake.

I don't think we need to belabor this (and it's not so important in the scheme of things) – just wanted to convey where I was coming from.